### PR TITLE
HVG-904: Restart embark from more options

### PR DESCRIPTION
**Pull request by hugokallstrom**
Any idea how we can simulate a press on the overflow menu button in the progress toolbar? Couldn't figure out how to do that, which is need if we want to write an integration test for this.

### Checklist

- [ ] Functionality is covered by an integration test
- [ ] Functionality is accessible in engineering mode

### CODE REVIEW

File: app/src/main/java/com/hedvig/app/feature/embark/ui/EmbarkActivity.kt [hunk 1]
```diff
@@ -27,26 +26,20 @@ import com.hedvig.app.feature.embark.passages.textactionset.TextActionSetFragmen
...
+        // TODO: Implement error UI that design must provide
+        intent.getStringExtra(STORY_NAME) ?: throw IllegalArgumentException("Programmer error: STORY_NAME not provided to ${this.javaClass.name}")
+    }
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
 
-        val storyName = intent.getStringExtra(STORY_NAME)
-
-        if (storyName == null) {
```
Comment (hugokallstrom) [inline app/src/main/java/com/hedvig/app/feature/embark/ui/EmbarkActivity.kt:42]: What is the reason we are checking the parameter in oncreate? I think it makes more sense to check this before starting the embark activity and showing an error dialog in the previous activity.

Comment (hedvigoscar) [reply app/src/main/java/com/hedvig/app/feature/embark/ui/EmbarkActivity.kt:42]: Just generally defensive programming, I think 🤔

File: app/src/main/java/com/hedvig/app/feature/embark/ui/MoreOptionsActivity.kt [hunk 2]
```diff
@@ -43,6 +44,10 @@ class MoreOptionsActivity : BaseActivity(R.layout.activity_more_options) {
                 showLogin()
             }
 
+            restart.setOnClickListener {
```
Comment (hedvigoscar) [inline app/src/main/java/com/hedvig/app/feature/embark/ui/MoreOptionsActivity.kt:47]:
```suggestion
            restart.setHapticClickListener {
```

File: app/src/main/java/com/hedvig/app/feature/embark/ui/EmbarkActivity.kt [hunk 3]
```diff
@@ -181,9 +174,18 @@ class EmbarkActivity : BaseActivity(R.layout.activity_embark) {
         }
     }
 
+    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+        if (requestCode == REQUEST_MORE_OPTIONS && resultCode == RESULT_CANCELED) {
+            model.load(storyName)
```
Comment (hedvigoscar) [inline app/src/main/java/com/hedvig/app/feature/embark/ui/EmbarkActivity.kt:179]: Should we flush the `store` at this point? 🤔

**Review by hedvigoscar — COMMENTED** [review]
One thought before we proceed 🙂

File: app/src/main/java/com/hedvig/app/feature/embark/ui/EmbarkActivity.kt (hunk 3 above)
Comment (hugokallstrom) [reply app/src/main/java/com/hedvig/app/feature/embark/ui/EmbarkActivity.kt:179]: Thought about that as well, I'll take it with Zak

**Review by hedvigoscar — APPROVED** [review]

**Comment by hedvigoscar** [comment]
Test are failing though, adress that before merging 🙂

**Pull request merged**